sync/sync_head.go

```diff
@@ -38,7 +38,11 @@ func (s *Syncer[H]) Head(ctx context.Context, _ ...header.HeadOption[H]) (H, err
 		return s.Head(ctx)
 	}
 	defer s.getter.Unlock()
-	netHead, err := s.getter.Head(ctx, header.WithTrustedHead[H](sbjHead))
+
+	reqCtx, cancel := context.WithTimeout(ctx, time.Second*2)
+	defer cancel()
+
+	netHead, err := s.getter.Head(reqCtx, header.WithTrustedHead[H](sbjHead))
```

Comment:
It would also be nice to check for ContextDeadline errors and change the log accordingly. Like, I couldn't get a recent header in time, returning the current subjective.

Reply:
I think having a single log is enough here
`log.Warnw("failed to get recent head, returning current subjective", "sbjHead", sbjHead.Height(), "err", err)`